etna/analysis/plotters.py

```diff
@@ -1312,12 +1322,61 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
     return holidays_df
 
 
+@_create_holidays_df.register
+def _create_holidays_df_dataframe(holidays: pd.DataFrame, index, as_is):
+    holidays_df = pd.DataFrame(index=index, columns=holidays["holiday"].unique(), data=0)
+
+    if as_is:
+        dt = holidays_df.index.intersection(holidays.index)
+        holidays_df.loc[dt, :] = holidays.loc[dt, :]
+        return holidays_df
+
+    for name in holidays["holiday"].unique():
+        ds = holidays[holidays["holiday"] == name]["ds"]
+        dt = [ds]
+        if "upper_window" in holidays.columns:
+            ds_upper_bound = ds + pd.to_timedelta(
+                holidays[holidays["holiday"] == name]["upper_window"], unit=index.freqstr
+            )
+            i = 0
+            while True:
+                ds_add = ds + pd.to_timedelta(
+                    holidays[holidays["holiday"] == name]["upper_window"] - i, unit=index.freqstr
+                )
+                case_up = ds_add <= ds_upper_bound
+                case_down = ds < ds_add
+                if not all(case_up & case_down):
+                    break
+                dt.append(ds_add)
+                i += 1
+        if "lower_window" in holidays.columns:
+            ds_lower_bound = ds - pd.to_timedelta(
+                holidays[holidays["holiday"] == name]["lower_window"], unit=index.freqstr
+            )
+            i = 0
+            while True:
+                ds_add = ds - pd.to_timedelta(
+                    holidays[holidays["holiday"] == name]["lower_window"] - i, unit=index.freqstr
+                )
+                case_down = ds_lower_bound <= ds_add
+                case_up = ds_add < ds
+                if not all(case_up & case_down):
+                    break
+                dt.append(ds_add)
+                i += 1
+        dt = pd.concat(dt)
+        dt = holidays_df.index.intersection(dt)
+        holidays_df.loc[dt, name] = 1
+    return holidays_df
+
+
 def plot_holidays(
```

Comment:
We should add Raises block for error with as_is.

Reply:
What do you mean? 
I've done it
`raise ValueError("Parameter `as_is` should be used with `holiday`: pd.DataFrame, not string.")`

Reply:
I mean Raises block in documentation:
```
Raises
------
ValueError:
    error description
```